### PR TITLE
EN-54151: Perform rollup validation using soql analyzer

### PR DESCRIPTION
**Pull request by chitang**
This validation was non-existent until recently with the introduction of compound query rollup.
Validation in DC requires complicated logic to handle datasets in different truth.  SF has all information at its fingertip.

### CODE REVIEW

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    val name = cid.underlying
+    name(0) match {
+      case ':' => new ColumnName(name)
+      case _ => new ColumnName("_" + name)
+    }
+  }
+
+  private def toDatasetContext(datasetRecord: DatasetRecord): com.socrata.soql.environment.DatasetContext[SoQLType] = {
+    val columnTypes = datasetRecord.columns.map { column =>
+      (column.fieldName, column.typ)
```
Comment (sjzavaleta) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala:103]: at this point, fieldName has "_" prefix if necessary?

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    dsRecord.columnsByName.mapValues(col => rollupColumnNameToIdMapping(col.id))
+  }
+
+  private def reverseColumnNameMap(dsRecord: MinimalDatasetRecord): Map[ColumnName, ColumnName] = {
+    dsRecord.columns.map { columnRecord =>
+      (rollupColumnNameToIdMapping(columnRecord.id), columnRecord.fieldName)
+    }.toMap
+  }
+
+  private def rollupColumnNameToIdMapping(cid: ColumnId): ColumnName = {
```
Comment (sjzavaleta) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala:93]: this seems like its columnIdToName, yeah?

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import com.socrata.soql.mapping.ColumnNameMapper
+import com.socrata.soql.parsing.{AbstractParser, Parser}
+import com.socrata.soql.types.SoQLType
+import com.socrata.soql.{BinaryTree, Compound, Leaf, SoQLAnalyzer}
+
+
+object RollupHelper {
+
+  private val soqlAnalyzer: SoQLAnalyzer[SoQLType] = new SoQLAnalyzer(SoQLTypeInfo, SoQLFunctionInfo)
+
```
[the remark this answers is not in the document]
Comment (chitang) [reply soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala:19]: removed

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala (hunk 1 above)
Comment (chitang) [reply soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala:102]: Unlike the one we had in DC.  We don't need prefix "_" here.  DC is dealing with a transformed soql while SF is dealing with the user soql before transform.

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala (hunk 2 above)
Comment (chitang) [reply soda-fountain-lib/src/main/scala/com/socrata/soda/server/highlevel/RollupHelper.scala]: I think you are right.  Renamed

**Review by sjzavaleta — APPROVED** [review]

**Pull request merged**